test/integration/tlspolicy_controller_test.go

```diff
@@ -84,7 +95,7 @@ var _ = Describe("TLSPolicy", Ordered, func() {
 					return k8sClient.Get(ctx, client.ObjectKey{Name: gateway.Name, Namespace: gateway.Namespace}, gateway)
 				}, TestTimeoutMedium, TestRetryIntervalMedium).ShouldNot(HaveOccurred())
 				tlsPolicy = NewTestTLSPolicy("test-tls-policy", testNamespace).
-					WithTargetGateway(gateway.Name).TLSPolicy
+					WithTargetGateway(gateway.Name).WithIssuer("testissuer", certmanv1.IssuerKind, "cert-manager.io").TLSPolicy
```

Comment:
nit: Subjective, but when using a builder pattern like this i think it looks a bit neater, and easier to understand, when each piece is on a new line:

```
tlsPolicy = NewTestTLSPolicy("test-tls-policy", testNamespace).
					WithTargetGateway(gateway.Name).
					WithIssuer("testissuer", certmanv1.IssuerKind, "cert-manager.io").TLSPolicy
```